File: 07. JavaScript ExpressJS Overview/Calculator/models/Calculator.js
function Calculator(leftOperand, operator, rightOperand) {
    this.leftOperand = leftOperand;
    this.operator = operator;
    this.rightOperand = rightOperand;

    this.calculateResult = function () {
        let result = 0;

        switch (this.operator) {
            case '+': result = this.leftOperand + this.rightOperand; break;
            case '-': result = this.leftOperand - this.rightOperand; break;
            case '*': result = this.leftOperand * this.rightOperand; break;
            case '/': result = this.leftOperand / this.rightOperand; break;
            case '^':
                result = 1;

                for(let i = 0; i < this.rightOperand; i++){
                    result *= this.leftOperand;
                }
                break;
            case 'AND': result = this.leftOperand & this.rightOperand; break;
            case 'OR': result = this.leftOperand | this.rightOperand; break;
            case 'XOR': result = this.leftOperand ^ this.rightOperand; break;
        }

        return result;
    }
}

module.exports = Calculator;

File: 09. Java Setup Spring Handle Forms/src/main/java/com/softuni/entity/Calculator.java
package com.softuni.entity;

public class Calculator {
    private double leftOperand;
    private double rightOperand;
    private String operator;

    public Calculator(double leftOperand, double rightOperand, String operator) {
        this.leftOperand = leftOperand;
        this.rightOperand = rightOperand;
        this.operator = operator;
    }

    public double getLeftOperand() {
        return leftOperand;
    }

    public void setLeftOperand(double leftOperand) {
        this.leftOperand = leftOperand;
    }

    public double getRightOperand() {
        return rightOperand;
    }

    public void setRightOperand(double rightOperand) {
        this.rightOperand = rightOperand;
    }

    public String getOperator() {
        return operator;
    }

    public void setOperator(String operator) {
        this.operator = operator;
    }

    public double calculateResult(){
        double result;

        switch (this.operator){
            case "+":
                result = this.getLeftOperand() + this.getRightOperand();
                break;
            case "-":
                result = this.getLeftOperand() - this.getRightOperand();
                break;
            case "*":
                result = this.getLeftOperand() * this.getRightOperand();
                break;
            case "/":
                if(this.getRightOperand() == 0){
                    throw new IllegalArgumentException("Cannot divide by zero!");
                }
                result = this.getLeftOperand() / this.getRightOperand();
                break;
            case "^":
                result = 1;
                for (int i = 1; i <= this.getRightOperand(); i++) {
                    result *= this.getLeftOperand();
                }
                break;
            case "AND":
                result = (int)this.getLeftOperand() & (int)this.getRightOperand();
                break;
            case "OR":
                result = (int)this.getLeftOperand() | (int)this.getRightOperand();
                break;
            case "XOR":
                result = (int)this.getLeftOperand() ^ (int)this.getRightOperand();
                break;
            default:
                result = 0;
                break;
        }

        return result;
    }
}
